core/src/net/sf/openrocket/aerodynamics/BarrowmanCalculator.java

```diff
@@ -477,7 +481,7 @@ private double calculateFrictionDrag(FlightConfiguration configuration, FlightCo
 		}
 		
 		// fB may be POSITIVE_INFINITY, but that's ok for us
-		double fB = (len + 0.0001) / maxR;
+		double fB = (maxX - minX + 0.0001) / maxR;
```

Comment:
Does this affect when the overall CD is overridden for the rocket? The changes will cause this to go from 0.0001 / maxR to Double.MAX / maxR. The correction applied to each of the components will be significantly different in this scenario, and I'm just not positive how its used.

Reply:
The problem I was trying to address was in the calculation of the length of the rocket -- the old code simply summed up the lengths of all the axially symmetric components, which had problems when there were pods:  if there was a central body tube and a pod off to the side, the lengths would be added together so len would be too long.  This code simply replaces that sum with the max and min x encountered while processing the rocket.
Unless I'm missing something, this shouldn't make any difference when the overall CD is overridden.

Reply:
Actually, I was wrong. Regardless of whether the overall CD being overridden or not, this value doesn't actually change/affect the computations. The key is that maxR is set to 0, so fb will always result in Double.INFINITY when the overall CD is overridden. The correction that's used will then always be 1 + 1/(2 * INFINITY) which will always be 1.0. So regardless of the change, it doesn't affect the correction value applied.

Apologies for the noise!